src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -136,11 +141,38 @@ private void subscribeToConfigChanges() {
     }
 
     private void onConfigurationChanged() {
-        try {
-            cdaConfiguration = CDAConfiguration.from(cdaConfiguration, getConfig());
-        } catch (URISyntaxException e) {
-            serviceErrored(e);
+        // Note: The nucleus emits multiple configuration changed events, one per key that changed. It will also
+        // keep emitting them regardless of the state it is current in. If the configuration was incorrect, we want the
+        // service to error, but we don't want to check again until the nucleus has run the remediation steps (when the
+        // service errors, the nucleus will try to call shutdown -> install -> startup).
```

Comment:
Can you help me understand this? My mental model is that we shouldn't need to care about what Nucleus is doing, and that we can rely entirely on configuration updates. If we are broken, and a configuration change fixes us, then request reinstall? Are there scenarios where that will not work?

Reply:
@jbutler and I chatted about this offline - we are going to dig deeper and see if it is required